[O11y][Apache Spark] Update README to use documentation guidelines (#9068)
elastic/integrations: packages/apache_spark/_dev/build/docs/README.md

## Data streams

The Apache Spark integration collects metrics data.

Metrics give you insight into the statistics of the Apache Spark. The `Metric` data streams collected by the Apache Spark integration are `application`, `driver`, `executor` and `node` so that the user can monitor and troubleshoot the performance of the Apache Spark instance.

Data streams:
- `application`: Collects information related to the number of cores used, application name, runtime in milliseconds and current status of the application.
- `driver`: Collects information related to the driver details, job durations, task execution, memory usage, executor status and JVM metrics.
- `executor`: Collects information related to the operations, memory usage, garbage collection, file handling, and threadpool activity.
- `node`: Collects information related to the application count, waiting applications, worker metrics, executor count, core usage and memory usage.

Note:
- Users can monitor and see the metrics inside the ingested documents for PHP-FPM in the `metrics-*` index pattern from `Discover`.

## Compatibility

This integration has been tested against `Apache Spark version 3.5.0`.

## Prerequisites

> You need Elasticsearch for storing and searching your data and Kibana for visualizing and managing it. You can use our hosted Elasticsearch Service on Elastic Cloud, which is recommended or self-manage the Elastic Stack on your own hardware.

Reviewer: I would suggest to:
1. Move this content to the section "Requirements"
2. Remove the section Prerequisites.
3. Put a comma after the text `which is recommended`

> Follow the same set of steps for Spark Worker, Driver and Executor.

Reviewer: Are these steps that we can link to?

Author: The steps, mentioned is only applicable for Apache Spark integration. Also user need to do repetitive steps to achieve the metrics for Worker, Driver and Executor. IMO, if we repeat the steps, the README will be getting redundant as well as large. WDYT?

Reviewer: Agreed, no need to repeat. I was just trying to figure out where these steps are, and maybe reference them.

## Setup

For step-by-step instructions on how to set up an integration, see the [Getting Started](https://www.elastic.co/guide/en/welcome-to-elastic/current/getting-started-observability.html) guide.

## Validation

> After the integration is successfully configured, clicking on the Assets tab of the Apache Spark Integration should display a list of available dashboards. Click on the dashboard available for your configured data stream. It should be populated with the required data.

Reviewer: ```suggestion
After the integration is successfully configured, click on the *Assets* tab of the Apache Spark Integration to display the available dashboards. Select the dashboard for your configured data stream, which should be populated with the required data.
```

### Troubleshooting

If `host.ip` appears conflicted under the ``metrics-*`` data view, this issue can be resolved by [reindexing](https://www.elastic.co/guide/en/elasticsearch/reference/current/tsds-reindex.html) the ``Application``, ``Driver``, ``Executor`` and ``Node`` data stream.

## Metrics

### Application

> This is the `application` data stream. This data stream gives metrics related to the number of cores used, application name, runtime in milliseconds and current status of the application.

Reviewer: ```suggestion
The `application` data stream provides metrics related to the number of cores used, application name, runtime in milliseconds, and current status of the application.

```

{{event "application"}}

{{fields "application"}}

### Driver

> This is the `driver` data stream. This data stream gives metrics related to the driver details, job durations, task execution, memory usage, executor status and JVM metrics.

Reviewer: ```suggestion
The `driver` data stream provides metrics related to the driver details, job durations, task execution, memory usage, executor status, and JVM metrics.

```

{{event "driver"}}

{{fields "driver"}}

### Executor

> This is the `executor` data stream. This data stream gives metrics related to the operations, memory usage, garbage collection, file handling, and threadpool activity.

Reviewer: ```suggestion
The `executor` data stream provides metrics related to the operations, memory usage, garbage collection, file handling, and threadpool activity.

```

{{event "executor"}}

{{fields "executor"}}

### Node

> This is the `node` data stream. This data stream gives metrics related to the application count, waiting applications, worker metrics, executor count, core usage and memory usage.

Reviewer: ```suggestion
The `node` data stream provides metrics related to the application count, waiting applications, worker metrics, executor count, core usage, and memory usage.

```
